src/CHIPBindings.cc

```diff
@@ -2472,10 +2477,14 @@ hipError_t hipHostRegister(void *HostPtr, size_t SizeBytes,
 hipError_t hipHostUnregister(void *HostPtr) {
   CHIP_TRY
   CHIPInitialize();
-  NULLCHECK(HostPtr);
+  if (!HostPtr)
+    CHIPERR_LOG_AND_THROW("Host pointer is nullptr!", hipErrorInvalidValue);
 
-  auto Device = Backend->getActiveDevice();
-  auto AllocInfo = Device->AllocationTracker->getAllocInfo(HostPtr);
+  auto *Device = Backend->getActiveDevice();
+  auto *AllocInfo = Device->AllocationTracker->getAllocInfo(HostPtr);
```

Comment:
What's the point of `auto *`? Doesn't auto automatically deduce the full type no matter if it's a pointer or not?

Reply:
For avoiding accidental copies ([link](https://llvm.org/docs/CodingStandards.html#beware-unnecessary-copies-with-auto)).